src/featuredetection/regexp.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module utils/featuredetection/regexp
```

Comment:
Does it have to be entire namespace for regexp? Do we expect to have many of those?

Reply:
I think that moving this to `env.js` will be cleaner.

Reply:
> Does it have to be entire namespace for regexp? Do we expect to have many of those?

@Reinmar There are several features that are about to come (not widely supported) into JS RegExp:

* s (a.k.a. dotAll) flag
* named groups
* Unicode property escapes (that's in this PR)

Also ES2018 finally added lookahead/lookbehind, luckily both seems to be implemented by all major browser vendors. 🎉

Note that future ES revisions might also expose more RegExp features to further reduce the gap.

Reply:
> There are several features that are about to come (not widely supported) into JS RegExp: (..)

I wonder how many of them are we going to use and will have to detect.

For me having feature detection inside `env.js` seems to be a good solution. The first example is more appealing to me.

Reply:
@mlewand What would be the problem if we had like 5 regexp-oriented properties in `env`?

We can create new namespaces for every new distinctive thing that may become "plural" with time. But many of them don't become plural and majority doesn't exceed 3 options. If you don't see technical problems for keeping that in `env`, I'd keep that in `env`. They will be easier to find there.

Reply:
@jodator 

```
       // or just
	features: {
		isRegExpUnicodePropertySupported: isRegExpUnicodePropertySupported()
	}
```

sounds best to me – it's the most flexible option.